src/main/java/org/folio/circulation/domain/validation/UserManualBlocksValidator.java

```diff
@@ -0,0 +1,66 @@
+package org.folio.circulation.domain.validation;
+
+import static org.folio.circulation.support.Result.of;
+import static org.folio.circulation.support.ValidationErrorFailure.singleValidationError;
+
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+import org.folio.circulation.domain.Request;
+import org.joda.time.DateTime;
+
+import org.folio.circulation.domain.ManualBlock;
+import org.folio.circulation.domain.MultipleRecords;
+import org.folio.circulation.domain.RequestAndRelatedRecords;
+import org.folio.circulation.domain.User;
+import org.folio.circulation.support.CollectionResourceClient;
+import org.folio.circulation.support.HttpFailure;
+import org.folio.circulation.support.MultipleRecordFetcher;
+import org.folio.circulation.support.Result;
+import org.folio.circulation.support.http.server.ValidationError;
+
+public class UserManualBlocksValidator {
+  private final CollectionResourceClient manualBlocksStorageClient;
+
+  public UserManualBlocksValidator(CollectionResourceClient manualBlocksStorageClient) {
+    this.manualBlocksStorageClient = manualBlocksStorageClient;
+  }
+
+  public CompletableFuture<Result<RequestAndRelatedRecords>> refuseWhenUserIsBlocked(
+    RequestAndRelatedRecords requestAndRelatedRecords) {
+
+    final User requester = Optional.ofNullable(requestAndRelatedRecords.getRequest())
+      .map(Request::getRequester).orElse(null);
+
+    if (requester != null) {
+      final MultipleRecordFetcher<ManualBlock> fetcher = new MultipleRecordFetcher<>(
```

Comment:
Can it be extracted as a class field?

Reply:
Do you mean "a class field" in Item class?

Reply:
nope, in this class - `UserManualBlocksValidator`, it seems that this fetcher is immutable and does not depend on tenant, etc. So it make sense to create it once, instead of creating each time when the method is called.

Reply:
Changed